From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: aswin-mcw <aswin.kalies@multicorewareinc.com>
Date: Wed, 9 Sep 2026 13:53:36 +0530
Subject: [PATCH 108/109] Revert "include stat.h"

(cherry picked from commit 83abfe26c)

This reverts temp: include stat.h.  Without it, cygwin/stat.h pulls in
machine/types.h, which defines struct flock a second time and breaks the
cygserver build:

  machine/types.h:35:8: error: redefinition of struct flock
  make: *** [Makefile:1062: cygserver-sysv_shm.o] Error 1
---
 winsup/cygwin/include/cygwin/stat.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/winsup/cygwin/include/cygwin/stat.h b/winsup/cygwin/include/cygwin/stat.h
index 2e83f680b..7f1b61681 100644
--- a/winsup/cygwin/include/cygwin/stat.h
+++ b/winsup/cygwin/include/cygwin/stat.h
@@ -9,7 +9,7 @@ details. */
 
 #ifndef _CYGWIN_STAT_H
 #define _CYGWIN_STAT_H
-#include <machine/types.h> 
+
 #ifdef __cplusplus
 extern "C" {
 #endif
